### ISSUE TITLE
RISCV baremetal toolchain purecap compiler rt assertion

### ISSUE THREAD

**Issue by arichardson** [CTSRD-CHERI/llvm-project#374]
```
/Users/alex/cheri/output/sdk/bin/clang -target riscv64-none-none-purecap -pipe -B/Users/alex/cheri/output/sdk/bin -mno-relax -Qunused-arguments -g -Wall -Werror -ffreestanding -nostdlib -fno-strict-aliasing -O2 -fno-omit-frame-pointer -fno-optimize-sibling-calls -mno-save-restore -mstrict-align -mabi=lp64 -march=rv64imafdc -mcmodel=medany -I/Users/alex/cheri/opensbi/platform/qemu/virt/include -I/Users/alex/cheri/opensbi/include -I/Users/alex/cheri/opensbi/lib/utils/libfdt/   -DFW_TEXT_START=0x80000000 -DFW_JUMP_ADDR=0x80200000 -DFW_JUMP_FDT_ADDR=0x82200000 -DFW_PAYLOAD_PATH=\"/Users/alex/cheri/build/opensbi-baremetal-riscv64-purecap-128-build/platform/qemu/virt/firmware/payloads/test.bin\" -DFW_PAYLOAD_OFFSET=0x200000 -DFW_PAYLOAD_FDT_ADDR=0x82200000 -DFW_OPTIONS=0x2   -fno-pie -no-pie /Users/alex/cheri/build/opensbi-baremetal-riscv64-purecap-128-build/platform/qemu/virt/firmware/payloads/test.o /Users/alex/cheri/build/opensbi-baremetal-riscv64-purecap-128-build/platform/qemu/virt/lib/libplatsbi.a -Wl,--build-id=none -Wl,--omagic   -Wl,-T/Users/alex/cheri/build/opensbi-baremetal-riscv64-purecap-128-build/platform/qemu/virt/firmware/payloads/test.elf.ld -o /Users/alex/cheri/build/opensbi-baremetal-riscv64-purecap-128-build/platform/qemu/virt/firmware/payloads/test.elf
 ELF       platform/qemu/virt/firmware/payloads/test.elf
Assertion failed: (Triple.getSubArch() != llvm::Triple::NoSubArch && "purecap triple should have subarch"), function getArchNameForCompilerRTLib, file /Users/alex/cheri/llvm-project/clang/lib/Driver/ToolChain.cpp, line 365.
```

**Comment by arichardson** [comment]
I've removed the assert.

**Comment by jrtc27** [comment]
Please stop using -purecap for RISC-V. I know why you want to do it, but it is not the RISC-V way of doing things.

**Comment by arichardson** [comment]
I am not using -purecap... But the assert only makes sense for MIPS so I guarded it with .isMIPS().

**Comment by arichardson** [comment]
And if -purecap is so evil, maybe the RISCV backend should reject it.

**Comment by jrtc27** [comment]
> /Users/alex/cheri/output/sdk/bin/clang -target riscv64-none-none-purecap ...

Or do you mean that you’re no longer using -purecap? But yes, I should reject it in the driver and backend, much like I assume is done for the weird ones like gnuabi64.

**Comment by arichardson** [comment]
Yes the -purecap was only there because I hadn't adjusted cheribuild for RISCV yet when I opened that bug. It now passes the right -march+-mabi flags.
If the MC layer now knows about the ABI everywhere we can remove the whole -purecap stuff for MIPS too. The only reason I added it is that there were lots of places where the only thing I had available was the target triple (e.g. when selecting the right dwarf register for stack pointer, return address etc.).
